action.yml

```diff
@@ -0,0 +1,28 @@
+name: Rust Marker Linter
+description: GitHub Action to install and run Marker linting for Rust 🦀
+branding:
+  icon: edit-3
+  color: white
+
+inputs:
+  install-only:
+    description: >
+      If set to `true` then the action will only install `cargo marker`,
+      and will skip running `cargo marker`. Use this if you want to run
+      something more complex than just `cargo marker`. If you think there
+      may be a frequent use case for running a different command then we will be
+      glad if you open a feature request issue for that to extend the action input
+      parameters.
+
+    default: 'false'
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - run: ${GITHUB_ACTION_PATH:?}/scripts/release/install.${{ runner.os == 'Windows' && 'ps1' || 'sh' }}
```

Comment:
Since this doesn't need to download the script, I wonder if it "clones" the entire repo for perform the composition. That would explain the advice, to have the action separate from the main repository :thinking: 

The Marker repo is currently about 7mb, which is reasonable, and would be fine for now. But it would be a good thing to know. Do you have any information about this? Or can we maybe test this, by simply running "ls"?

Reply:
I suppose it does a [sparse-checkout](https://git-scm.com/docs/git-sparse-checkout) or smth similar. I'll check what the file tree there is, and if there even is any `.git` dir there. Even if it's 7MB, I think it's nothing compared to downloading the pinned nightly rust toolchain (although with `--profile minimal`), which when decompressed is `544MB`. 
But that's something to keep in mind, definitely